x/oracle/simulations/operations.go

```diff
@@ -0,0 +1,321 @@
+// the aggregated exchange rates.
+func GenerateExchangeRatesString(prices map[string]sdk.Dec) string {
+	exchangeRates := make([]string, len(prices))
+	i := 0
+
+	// aggregate exchange rates as "<base>:<price>"
+	for base, avgPrice := range prices {
+		exchangeRates[i] = fmt.Sprintf("%s:%s", base, avgPrice.String())
+		i++
+	}
+
+	sort.Strings(exchangeRates)
+
+	return strings.Join(exchangeRates, ",")
+}
+
+// WeightedOperations returns all the operations from the module with their respective weights
+func WeightedOperations(
+	appParams simtypes.AppParams,
+	cdc codec.JSONCodec,
+	ak types.AccountKeeper,
+	bk types.BankKeeper,
+	k keeper.Keeper) simulation.WeightedOperations {
+	var (
+		weightMsgAggregateExchangeRatePrevote int
+		weightMsgAggregateExchangeRateVote    int
+		weightMsgDelegateFeedConsent          int
+	)
+	appParams.GetOrGenerate(cdc, OpWeightMsgAggregateExchangeRatePrevote, &weightMsgAggregateExchangeRatePrevote, nil,
+		func(_ *rand.Rand) {
+			weightMsgAggregateExchangeRatePrevote = simappparams.DefaultWeightMsgSend * 2
+		},
+	)
+
+	appParams.GetOrGenerate(cdc, OpWeightMsgAggregateExchangeRateVote, &weightMsgAggregateExchangeRateVote, nil,
+		func(_ *rand.Rand) {
+			weightMsgAggregateExchangeRateVote = simappparams.DefaultWeightMsgSend * 2
+		},
+	)
+
+	appParams.GetOrGenerate(cdc, OpWeightMsgDelegateFeedConsent, &weightMsgDelegateFeedConsent, nil,
+		func(_ *rand.Rand) {
+			weightMsgDelegateFeedConsent = simappparams.DefaultWeightMsgSetWithdrawAddress
+		},
+	)
+
+	return simulation.WeightedOperations{
+		simulation.NewWeightedOperation(
+			weightMsgAggregateExchangeRatePrevote,
+			SimulateMsgAggregateExchangeRatePrevote(ak, bk, k),
+		),
+		simulation.NewWeightedOperation(
+			weightMsgAggregateExchangeRateVote,
+			SimulateMsgAggregateExchangeRateVote(ak, bk, k),
+		),
+		simulation.NewWeightedOperation(
+			weightMsgDelegateFeedConsent,
+			SimulateMsgDelegateFeedConsent(ak, bk, k),
+		),
+	}
+}
+
+// SimulateMsgAggregateExchangeRatePrevote generates a MsgAggregateExchangeRatePrevote with random values.
+func SimulateMsgAggregateExchangeRatePrevote(
+	ak types.AccountKeeper,
+	bk types.BankKeeper,
+	k keeper.Keeper,
+) simtypes.Operation {
+	return func(
+		r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simtypes.Account, chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		simAccount, _ := simtypes.RandomAcc(r, accs)
+		address := sdk.ValAddress(simAccount.Address)
+		// ensure the validator exists
+		val := k.StakingKeeper.Validator(ctx, address)
+		if val == nil || !val.IsBonded() {
+			return simtypes.NoOpMsg(
+				types.ModuleName,
+				types.TypeMsgAggregateExchangeRatePrevote,
+				"unable to find validator",
+			), nil, nil
+		}
+
+		prices := make(map[string]sdk.Dec, len(acceptList))
+		for _, denom := range acceptList {
+			prices[denom] = simtypes.RandomDecAmount(r, sdk.NewDec(1)).Add(umeePrice)
```

Comment:
we should use `r` to generate a random value [0,1] and add that to `umeePrice` IMO.